include/velocypack/Slice.h

```diff
@@ -117,38 +117,70 @@ class Slice {
   // creates a slice of type MaxKey
   static constexpr Slice maxKeySlice() noexcept { return Slice(maxKeySliceData); }
 
-  // pointer to the head byte
-  constexpr uint8_t const* start() const noexcept { return _start; }
-  
+  int64_t getTagId() const {
+    if(isTagged()) {
+      if(*_start == 0xee) {
+        return readIntegerFixed<ValueLength, 1>(_start + 1);
+      } else if(*_start == 0xef) {
+        return readIntegerFixed<ValueLength, 8>(_start + 1);
+      } else {
+        throw new Exception(Exception::InternalError, "Invalid tag type ID");
+      }
+    }
+
+    return 0;
+  }
+
+  constexpr uint8_t tagOffset() const {
+    return tagOffset(rawStart());
+  }
+
+  // pointer to the head byte, including possible tags
+  constexpr uint8_t const* rawStart() const noexcept {
+    return _start;
+  }
+
+  // pointer to the head byte, excluding possible tags
+  constexpr uint8_t const* start() const noexcept {
```

Comment:
The `noexcept` is now misleading, as `tagOffset()` is potentially throwing.
I dislike removing the `noexcept` specifier from `start()`, as it is a very central and hot function, which is used in so many places in the library itself and in code that uses the library.

Additionally, this modification changes the behavior of `start()` so that it returns the VPack data without the tag. That means that embedders that use `start()` will lose all the tag information unless all embedder code is changed.
I think it should be the other way round: the behavior of `start()` should not be changed by this PR, instead there should be an additional (new) method that returns the VPack data without the tag. That will allow all embedder code to work the same way as before, as do a decoding of tags in the application, or, if not desired, call the new method to get the raw VPack data without the data.
But in any case, the behavior of `start()` should not be changed for downwards-compatibility reasons.